rag/easy-rag/runtime/src/main/java/io/quarkiverse/langchain4j/easyrag/runtime/EasyRagRecorder.java

```diff
@@ -35,37 +40,84 @@ public void ingest(EasyRagConfig config, BeanContainer beanContainer) {
+            // When the InMemoryEmbeddingStore bean was created
+            // See the inMemoryEmbeddingStoreSupplier method
+            if (!Files.exists(embeddingsFile)) {
+                // embeddingsFile doesn't exist, so ingest the documents and then write out the results
+                try {
+                    Files.createDirectories(embeddingsFile.getParent());
+                } catch (IOException ex) {
+                    throw new RuntimeException(ex);
+                }
+
+                ingestDocumentsFromFilesystem(config, embeddingStore, embeddingModel);
+                LOGGER.infof("Writing embeddings to %s", embeddingsFile);
+                ((InMemoryEmbeddingStore<TextSegment>) embeddingStore).serializeToFile(embeddingsFile);
+            }
+        } else {
+            ingestDocumentsFromFilesystem(config, embeddingStore, embeddingModel);
+        }
+    }
+
+    private void ingestDocumentsFromFilesystem(EasyRagConfig config, EmbeddingStore<TextSegment> embeddingStore,
+            EmbeddingModel embeddingModel) {
         PathMatcher pathMatcher = FileSystems.getDefault().getPathMatcher(config.pathMatcher());
         LOGGER.info("Ingesting documents from path: " + config.path() +
                 ", path matcher = " + config.pathMatcher() + ", recursive = " + config.recursive());
-        List<Document> documents = null;
-        if (config.recursive()) {
-            documents = FileSystemDocumentLoader.loadDocumentsRecursively(config.path(), pathMatcher);
-        } else {
-            documents = FileSystemDocumentLoader.loadDocuments(config.path(), pathMatcher);
-        }
+
+        List<Document> documents = config.recursive()
+                ? FileSystemDocumentLoader.loadDocumentsRecursively(config.path(), pathMatcher)
+                : FileSystemDocumentLoader.loadDocuments(config.path(), pathMatcher);
+
         DocumentSplitter documentSplitter = DocumentSplitters.recursive(config.maxSegmentSize(),
                 config.maxOverlapSize(), new HuggingFaceTokenizer());
+
         List<Document> splitDocuments = documentSplitter
                 .splitAll(documents)
                 .stream()
                 .map(split -> new Document(split.text()))
                 .toList();
+
         EmbeddingStoreIngestor.builder()
                 .embeddingModel(embeddingModel)
                 .embeddingStore(embeddingStore)
                 .build()
                 .ingest(splitDocuments);
+
         LOGGER.info("Ingested " + documents.size() + " files as " + splitDocuments.size() + " documents");
     }
 
-    public Supplier<InMemoryEmbeddingStore> inMemoryEmbeddingStoreSupplier() {
+    private Path getReuseEmbeddingsFilePath(EasyRagConfig config) {
+        return config.reuseEmbeddings().file()
+                .map(String::trim)
+                .map(Paths::get)
+                .orElseThrow(() -> new ConfigValidationException(createConfigProblems("reuse-embeddings.file")));
+    }
```

Comment:
You know I very much dislike this, but in this case my main complaint is that the error message should state that the file should be specified because a certain feature has been turned on. 
I am also wondering if just using some temp file makes sense, it certainly would reduce the friction in using the feature

Reply:
Without using the map stuff, this method would look like

```java
private Path getReuseEmbeddingsFilePath(EasyRagConfig config) {
        if (config.reuseEmbeddings().file().isPresent()) {
            return Paths.get(config.reuseEmbeddings().file().get().trim());
        }

        throw new ConfigValidationException(createConfigProblems("reuse-embeddings.file"));
}
```

I can make it that way if you'd prefer. I think the `map`/`orElseThrow` is much more readable, but I guess thats personal opinion.

Reply:
> and the user wouldn't really know where the embeddings file was in case they wanted to examine and/or save it

Do you think they would really need to do this?

Reply:
> > and the user wouldn't really know where the embeddings file was in case they wanted to examine and/or save it
> 
> Do you think they would really need to do this?

If I were building a demo I'd want to commit the embeddings file into source control. Or maybe in my tests I configure it to use a file, and in that case I'd want to know where the file was so that I could create it.

> > I can make it that way if you'd prefer. I think the map/orElseThrow is much more readable, but I guess thats personal opinion.
>

> My main gripe if we keep the file property optional is that the user does not know why this new property is needed - the message does not help much

I could default it to something (like somewhere in `Files.createTempDirectory`) and that way the user can override if they'd like, or if not, they don't have to care?

Reply:
> If I were building a demo I'd want to commit the embeddings file into source control. Or maybe in my tests I configure it to use a file, and in that case I'd want to know where the file was so that I could create it and/or put in somewhere in src/test/resources and commit it

Fair enough. Then maybe it can default to the current working directory?

Reply:
> Fair enough. Then maybe it can default to the current working directory?

What would the current working directory actually be? I think its value would be different if the app was run in dev mode vs `java -jar` vs in native?

At least with temp dir we could have something thats deterministic, and we could print it out to the console.

Reply:
> What would the current working directory actually be? I think its value would be different if the app was run in dev mode vs java -jar vs in native?

It depends on where you run it from, but with `mvn quarkus:dev` and `java -jar target/quarkus-app/quarkus-run.jar` and `./target/whatever-runner` it would be the root of the project which seems like a reasonable choice.

In any case, it's not terribly important and we don't need to spend too much time finding the best default. I trust you'll use whatever you find best by working with the feature.

Reply:
just pushed new changes